spring-integration-redis/src/main/java/org/springframework/integration/redis/util/RedisLockRegistry.java

```diff
@@ -207,6 +216,17 @@ public void setExecutor(Executor executor) {
 		this.executorExplicitlySet = true;
 	}
 
+	/**
+	 * Set the {@link TaskScheduler} to use for the renewal task.
+	 * when renewalTaskScheduler is set, it will be used to periodically renew the lock to ensure that the lock does not expire while the thread is working.
+	 * @param renewalTaskScheduler renew task scheduler
+	 * @since 6.4.0
+	 */
+	@Override
+	public void setRenewalTaskScheduler(TaskScheduler renewalTaskScheduler) {
+		this.renewalTaskScheduler = renewalTaskScheduler;
```

Comment:
`Assert.notNull(renewalTaskScheduler, "'renewalTaskScheduler' must not be null")`